warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.log4j.Logger;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.Multimap;
+
+import datawave.ingest.data.Type;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.ingest.data.config.NormalizedFieldAndValue;
+
+public class WhindexFieldIngestHelper implements WhindexIngest {
+
+    private static final Logger log = Logger.getLogger(WhindexFieldIngestHelper.class);
+
+    public static final String WHINDEX_RULES = "whindex.rules";
+    public static final String VALUE_FIELD = "value_field";
+    public static final String SRC_FIELD = "src_field";
+    public static final String DELETE_SRC_FIELD = "delete_src_field";
+    public static final String DST_FIELD = "dst_field";
+    public static final String VALUES = "values";
+
+    private final Type type;
+    private final Multimap<String,WhindexConfig> valueFieldsToWhindexConfigs = HashMultimap.create();
+    private final Set<String> overloadedFields = new HashSet<>();
+
+    public WhindexFieldIngestHelper(Type type) {
+        this.type = type;
+    }
+
+    @Override
+    public void setup(Configuration config) throws IllegalArgumentException {
+
+        String commonPrefix = type.typeName() + "." + WHINDEX_RULES + ".";
+
+        Map<String,String> properties = config.getPropsWithPrefix(commonPrefix);
+        Map<String,WhindexConfig> groupingsToConfigs = new HashMap<>();
+
+        for (Map.Entry<String,String> entry : properties.entrySet()) {
+            String[] parts = entry.getKey().split("\\.");
+            String groupID = parts[0];
+            String property = parts[1];
+
+            WhindexConfig whindexConfig = groupingsToConfigs.computeIfAbsent(groupID, (k) -> new WhindexConfig());
+            switch (property) {
+                case VALUE_FIELD:
+                    whindexConfig.setValueField(entry.getValue());
+                    break;
+                case SRC_FIELD:
+                    whindexConfig.setSourceField(entry.getValue());
+                    break;
+                case DELETE_SRC_FIELD:
+                    whindexConfig.setOverloaded(Boolean.parseBoolean(entry.getValue()));
+                    if (whindexConfig.isOverloaded()) {
+                        overloadedFields.add(whindexConfig.getSourceField());
+                    }
+                    break;
+                case DST_FIELD:
+                    whindexConfig.setDestField(entry.getValue());
+                    break;
+                case VALUES:
+                    whindexConfig.setValues(List.of(entry.getValue().split(",")));
+                    break;
+                default:
+                    String originalProperty = commonPrefix + groupID + "." + property;
+                    log.warn("Unexpected whindex property given:" + originalProperty + "=" + entry.getValue());
+            }
+        }
+
+        // add to fieldDefinitions and VFToWC
```

Comment:
This comment is not very clear, there is no variable called fieldDefinitions, and you should avoid abbreviating variable like you did for VFToWC in documentation. Please reference by the full variable name, or an understandable concept like "map of value fields to whindex configs".